plugin/evm/message/codec.go

```diff
@@ -15,7 +15,11 @@ const (
 	maxMessageSize = 2*units.MiB - 64*units.KiB // Subtract 64 KiB from p2p network cap to leave room for encoding overhead from AvalancheGo
 )
 
-var Codec codec.Manager
+var (
+	Codec codec.Manager
+	// TODO: Remove this once we have a better way to register types (i.e use a different codec version or use build flags)
```

Comment:
we can move init to a function like (Initialize) and call in vm.Initialize like `Codec.Initialize(SyncSummaryType)`, but then tests should be doing the same thing everytime. But it might be the only way once we merge subnet-evm since two VMs should be using two different codecs since subnet-evm do not have atomic types (and thus codec has different order of type IDs).